adaptive_breakpoints/lib/src/adaptive_breakpoints.dart

```diff
@@ -0,0 +1,377 @@
+        return 719;
+        break;
+      case AdaptiveWindow.m:
+        return 959;
+        break;
+      case AdaptiveWindow.l:
+        return 1279;
+        break;
+      case AdaptiveWindow.xl:
+        return 10000000;
+        break;
+      default:
+        throw AssertionError('Error');
+    }
+  }
+
+  double get shortestLandscapeHeight {
+    switch (this) {
+      case AdaptiveWindow.xs:
+        return 0;
+        break;
+      case AdaptiveWindow.s:
+        return 360;
+        break;
+      case AdaptiveWindow.m:
+        return 720;
+        break;
+      case AdaptiveWindow.l:
+        return 960;
+        break;
+      case AdaptiveWindow.xl:
+        return 1280;
+      default:
+        throw AssertionError('Error');
+    }
+  }
+
+  double get longestPortraitHeight {
+    switch (this) {
+      case AdaptiveWindow.xs:
+        return 959;
+        break;
+      case AdaptiveWindow.s:
+        return 1599;
+        break;
+      case AdaptiveWindow.m:
+        return 1919;
+        break;
+      case AdaptiveWindow.l:
+        return 1279;
+        break;
+      case AdaptiveWindow.xl:
+        return 10000000;
+        break;
+      default:
+        throw AssertionError('Error');
+    }
+  }
+
+  double get shortestPortraitHeight {
+    switch (this) {
+      case AdaptiveWindow.xs:
+        return 0;
+        break;
+      case AdaptiveWindow.s:
+        return 960;
+        break;
+      case AdaptiveWindow.m:
+        return 1600;
+        break;
+      case AdaptiveWindow.l:
+        return 960;
+        break;
+      case AdaptiveWindow.xl:
+        return 1280;
+      default:
+        throw AssertionError('Error');
+    }
+  }
+}
+
+class AdaptiveWindowLimits {
+  final int xs;
+  final int s;
+  final int m;
+  final int l;
+  final int xl;
+
+  AdaptiveWindowLimits({
+    @required this.xs,
+    @required this.s,
+    @required this.m,
+    @required this.l,
+    @required this.xl,
+  })  : assert(xs != null),
+        assert(s != null),
+        assert(m != null),
+        assert(l != null),
+        assert(xl != null);
+}
+
+/// This class represents the Material breakpoint system entry.
+/// https://material.io/design/layout/responsive-layout-grid.html#breakpoints
+class BreakpointSystemEntry {
+  /// The breakpoint range values use the height of the device if the user is
+  /// using their device on portrait view or the width of the device if the user
+  /// is using their device in landscape view.
+  /// By using this logic we can compare the breakpoint range values to the longer
+  /// length of the MediaQuery.
+  /// This is typical done by using: MediaQuery.of(context).size.longestSide
+  final RangeValues range;
+
+  /// String of device name that use this breakpoint range in portrait view.
+  final String portrait;
+
+  /// String of device name that use this breakpoint range in landscape view.
+  final String landscape;
+
+  /// Material generalizes the Device Size into five different windows: extra
```

Comment:
Needs clarification, since this doesn't explain what the window parameter is or does

Reply:
Done